tabular/src/autogluon/tabular/configs/pipeline_presets.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+import math
+
+from autogluon.core.constants import BINARY, PROBLEM_TYPES
+from autogluon.core.utils.utils import default_holdout_frac
+
+USE_BAG_HOLDOUT_AUTO_THRESHOLD = 1_000_000
+
+
+def _get_validation_preset(num_train_rows: int, hpo_enabled: bool) -> dict[str, int | float]:
+    """Recommended validation preset manually defined by the AutoGluon developers."""
+
+    # Default recommendation
+    num_bag_folds = 8  # due to 8 cores per CPU being very common.
+    num_bag_sets = 1
+
+    use_bag_holdout = num_train_rows >= USE_BAG_HOLDOUT_AUTO_THRESHOLD
+    holdout_frac = round(default_holdout_frac(num_train_rows=num_train_rows, hyperparameter_tune=hpo_enabled), 4)
+
+    # Changes to avoid overfitting
+    if num_train_rows <= 1_000:
+        num_bag_folds = min(8, max(4, math.floor(num_train_rows / 100)))
+        num_bag_sets = min(10, 4 + (2 * (8 - num_bag_folds)))
+    elif num_train_rows < 5_000:
+        num_bag_sets = 4
+    elif num_train_rows < 10_000:
+        num_bag_sets = 2
+    elif num_train_rows < USE_BAG_HOLDOUT_AUTO_THRESHOLD:
+        # TODO(improvement): go towards 4-fold cv for more than 100k rows?
+        pass
+    else:
+        # Only use holdout validation afterward
+        num_bag_folds = 0
+        num_bag_sets = 0
```

Comment:
interesting idea, however this would presumably disable stacking then. Maybe that is ok?

Reply:
`USE_BAG_HOLDOUT_AUTO_THRESHOLD` still uses bagging, it just enables `use_bag_holdout`, so this is a change in logic. Unsure if intended

Reply:
I intended to disable cross-validation, but I somehow wrongly assumed that `USE_BAG_HOLDOUT_AUTO_THRESHOLD` also disabled it. I did not consider that this also disables stacking.

My thought process was that the larger the number of rows, the fewer folds/bags we would need to avoid overfitting. Thus, no bagging would be needed once we hit a critical point. But currently, we do not support stacking without bagging. 

What was the original motivation for `USE_BAG_HOLDOUT_AUTO_THRESHOLD`? Avoid stacked overfitting in these cases or use less expensive validation.

Reply:
The original motivation was to avoid stack overfitting without needing dynamic stacking when the number of rows were large. The only motivation was that the bagged model strength would be better understood by the weighted ensemble when taking the bagged predictions as input, potentially leading to better weights.

Reply:
I see, I changed it such that we keep the old logic.